src/parallel/parallel.hpp

```diff
@@ -131,9 +132,13 @@ class SegmentLock {
 /// HELPER CLASSES
 ////////////////////////////////////////////////////////////////////////////////
 
-struct ParallelBucket {
+struct KeyValue {
     KeyType key = 0;
     ValueType value = 0;
+};
+
+struct ParallelBucket {
+    std::atomic<KeyValue> key_value;
```

Comment:
Nice. Is this guaranteed to use atomic operations (instead of locks) for up-to 16 bytes?

Reply:
@thedavidchu We can probably handle that with a `static_assert(std::atomic<KeyValue>::is_always_lock_free)`.